widgets/src/slider.rs

```diff
@@ -127,6 +127,18 @@ impl Slider {
         
         self.draw_slider.end(cx);
     }
+
+    pub fn value(&self) -> f64 {
+        self.to_external()
+    }
+
+    pub fn set_value(&mut self, v: f64) {
+        let prev_value = self.value();
+        self.set_internal(v);
+        if v != prev_value {
```

Comment:
If you are binding the value of this slider to something external, by calling `set_value`, then, we should not touch the text input unless the value being set is actually different.

This is because would disallow you to have a dirty/incomplete state in the text input, which means you may not be abale to edit the slider using the text input.

I'm pretty sure this would work as

```rust
pub fn set_value(&mut self, v: f64) {
    if v != self.value() {
        self.set_internal(v);
        self.update_text_input();
    }
}
```

But since most widgets always set their internal state unconditionally, I would like to avoid not matching that behavior here. So calling always `set_internal` and just focusing on not touching the text input.